src/peer-routing/index.ts

```diff
@@ -235,8 +235,6 @@ export class PeerRouting implements Initializable {
     }
 
     for await (const event of this.queryManager.run(key, tablePeers, getCloserPeersQuery, options)) {
-      yield event
```

Comment:
I would expect [this function](https://github.com/libp2p/js-libp2p/blob/111e75d05e41eb2141fefb2cfafee0937fe068c2/src/peer-routing.ts#L171) to yield the `kBucketSize` sorted peers, but it's yielding a lot more than expected (not all are `FINAL_PEER`), and not necessarily sorted (because it merges events yielded by both `lan` and `wan` dhts).